gpflow/optimizers/natgrad.py

```diff
@@ -39,114 +47,223 @@ class XiTransform(metaclass=abc.ABCMeta):
 
         :param nat1: the θ₁ parameter
         :param nat2: the θ₂ parameter
         :return: tuple `xi1`, `xi2`
         """
-        pass  # pragma: no cover
+
+
+class XiNat(XiTransform):
+    """
+    This is the default transform. Using the natural directly saves the forward mode
+    gradient, and also gives the analytic optimal solution for gamma=1 in the case
+    of Gaussian likelihood.
+    """
+
+    @staticmethod
+    def meanvarsqrt_to_xi(mean, varsqrt):
+        return meanvarsqrt_to_natural(mean, varsqrt)
+
+    @staticmethod
+    def xi_to_meanvarsqrt(xi1, xi2):
+        return natural_to_meanvarsqrt(xi1, xi2)
+
+    @staticmethod
+    def naturals_to_xi(nat1, nat2):
+        return nat1, nat2
+
+
+class XiSqrtMeanVar(XiTransform):
+    """
+    This transformation will perform natural gradient descent on the model parameters,
+    so saves the conversion to and from Xi.
+    """
+
+    @staticmethod
+    def meanvarsqrt_to_xi(mean, varsqrt):
+        return mean, varsqrt
+
+    @staticmethod
+    def xi_to_meanvarsqrt(xi1, xi2):
+        return xi1, xi2
+
+    @staticmethod
+    def naturals_to_xi(nat1, nat2):
+        return natural_to_meanvarsqrt(nat1, nat2)
 
 
 class NaturalGradient(tf.optimizers.Optimizer):
-    def __init__(self, gamma: Scalar, name=None):
+    """
+    Implements a natural gradient descent optimizer for variational models
+    that are based on a distribution q(u) = N(q_mu, q_sqrt q_sqrtᵀ) that is
+    parameterized by mean q_mu and lower-triangular Cholesky factor q_sqrt
+    of the covariance.
+
+    Note that this optimizer does not implement the standard API of
+    tf.optimizers.Optimizer. Its only public method is minimize(), which has
+    a custom signature (var_list needs to be a list of (q_mu, q_sqrt) tuples,
+    where q_mu and q_sqrt are gpflow.Parameter instances, not tf.Variable).
+
+    When using in your work, please cite
+
+        @inproceedings{salimbeni18,
+            title={Natural Gradients in Practice: Non-Conjugate Variational Inference in Gaussian Process Models},
+            author={Salimbeni, Hugh and Eleftheriadis, Stefanos and Hensman, James},
+            booktitle={AISTATS},
+            year={2018}
+    """
+
+    def __init__(self, gamma: Scalar, xi_transform: XiTransform = XiNat(), name=None):
+        """
+        :param gamma: natgrad step length
+        :param xi_transform: default ξ transform (can be overridden in the call to minimize())
+            The XiNat default choice works well in general.
+        """
         name = self.__class__.__name__ if name is None else name
         super().__init__(name)
         self.gamma = gamma
+        self.xi_transform = xi_transform
 
     def minimize(
-        self,
-        loss_fn: Callable,
-        var_list: List[
-            Union[Tuple[Parameter, Parameter], Tuple[Parameter, Parameter, XiTransform]]
-        ],
+        self, loss_fn: LossClosure, var_list: Sequence[NatGradParameters],
```

Comment:
i'm going to chime in about one thing cos I think it's important. This method signature isn't compatible with that of `tf.optimizer.Optimizer`. If that doesn't affect this PR, we can come back to it another time

Reply:
Yes, very true! When reworking the code for this PR that is something that I also thought of. That is why I tried to make it much clearer in the docstrings - let me know if you have any further suggestions for clarifying that! I'm not sure what a good solution for this could be, I'd appreciate suggestions. It's beyond the scope of this PR though - I haven't actually changed the interface of this method, it behaves exactly as before.

Reply:
perhaps it doesn't need to be a `tf.optimizers.Optimizer` and could just be its own class

to make it an actual `tf.optimizers.Optimizer`, I don't know how we'd do that. It may very well be possible but I couldn't say just yet